tools/cmock/create_test.cmake

```diff
@@ -39,7 +39,7 @@ function(create_test test_name
     foreach(dependency IN LISTS dep_list)
         add_dependencies(${test_name} ${dependency})
     endforeach()
-    target_link_libraries(${test_name} -lgcov)
+    target_link_libraries(${test_name} -lgcov -lunity)
```

Comment:
I need this change because some tests do not have any use for mocks like for `mqtt_state_utest` and `mqtt_lightweight_utest`.